presto-parquet/src/main/java/io/prestosql/parquet/reader/MetadataReader.java

```diff
@@ -314,55 +313,58 @@ private static PrimitiveTypeName getTypeName(Type type)
         }
     }
 
-    private static OriginalType getOriginalType(ConvertedType type)
+    // Copied from org.apache.parquet.format.converter.ParquetMetadataConverter#getLogicalTypeAnnotation(org.apache.parquet.format.ConvertedType, org.apache.parquet.format.SchemaElement)
+    private static LogicalTypeAnnotation getLogicalTypeAnnotation(ConvertedType type, SchemaElement schemaElement)
     {
         switch (type) {
             case UTF8:
-                return OriginalType.UTF8;
+                return LogicalTypeAnnotation.stringType();
             case MAP:
-                return OriginalType.MAP;
+                return LogicalTypeAnnotation.mapType();
             case MAP_KEY_VALUE:
-                return OriginalType.MAP_KEY_VALUE;
+                return LogicalTypeAnnotation.MapKeyValueTypeAnnotation.getInstance();
             case LIST:
-                return OriginalType.LIST;
+                return LogicalTypeAnnotation.listType();
             case ENUM:
-                return OriginalType.ENUM;
+                return LogicalTypeAnnotation.enumType();
             case DECIMAL:
-                return OriginalType.DECIMAL;
+                int scale = (schemaElement == null ? 0 : schemaElement.scale);
```

Comment:
Use the getters

Reply:
This is verbatim copy of `org.apache.parquet.format.converter.ParquetMetadataConverter#getLogicalTypeAnnotation`, I'd prefer to keep it this way.